include/ttmlir/Dialect/TT/IR/TTOpsTypes.td

```diff
@@ -235,6 +235,20 @@ def TT_LayoutAttr : TT_Attr<"Layout", "layout"> {
   }];
 }
 
+def TT_ShardLayoutAttr : TT_Attr<"ShardLayout", "shard_layout", []> {
```

Comment:
Yes, this will directly map to a metal shard.  A circular buffer will also be constructed as a view around this shard as means for tensix to access the data.

Reply:
I'd also be happy calling this `Buffer`, maybe shard is a bit too specific because I think in the future we need an attribute on this that specifies a `mode`, which is either:
- `alias`: Construct a shard/buffer around an existing tensor allocation in L1, this has lowering semantics of trivially constructing a circular buffer around the tensor, no additional data-movement needed.
- `stream`: Construct a shard/buffer around a temporary storage allocation in L1.  This has lowering semantics of gather tensor data from a remote source (DRAM or another core's L1) and the compiler would need to generate a data movement kernel for it.

Reply:
I ended up renaming to `Buffer`, lmk if that makes sense to you too.  I can change it back or to something else if you think.

Reply:
It's fine with me.